### PR TITLE
Update CI with linting and fix issues reported by golangci-lint

### PR DESCRIPTION
**Pull request by fasmat**

### CODE REVIEW

File: validation.go [hunk 1]
```diff
@@ -79,7 +85,7 @@ func (v *Validator) CalcRoot(stopAtLayer uint) ([]byte, []ParkingSnapshot, error
 		if activePos.Height == stopAtLayer {
 			break
 		}
-		// The activeNode's sibling should be calculated iff it's an ancestor of the next proven leaf. Otherwise, the
+		// The activeNode's sibling should be calculated if it's an ancestor of the next proven leaf. Otherwise, the
```
Comment (poszu) [inline validation.go:88]: I think this was intended. Iff stands for "if and only if".

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -8,9 +8,46 @@ on:
...
+        uses: actions/checkout@v4
+      - name: set up go
+        uses: actions/setup-go@v5
+        with:
+          check-latest: true
+          go-version-file: "go.mod"
+      - name: setup env
+        run: make install
+      - name: lint
+        run: make lint-github-action
```
Comment (poszu) [inline .github/workflows/ci.yml:41]: How about merging quicktests and lint into a single job?

File: validation.go (hunk 1 above)
Comment (fasmat) [reply validation.go:88]: hm... golangci-lint marks it as a typo 🤔

File: .github/workflows/ci.yml (hunk 2 above)
Comment (fasmat) [reply .github/workflows/ci.yml:41]: We could I wanted to keep the same structure as we have in other repos

Comment (poszu) [reply .github/workflows/ci.yml:41]: Sure, we can merge it everywhere :P Or leave it as is :shrug:

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -26,8 +63,38 @@ jobs:
         with:
           check-latest: true
           go-version-file: "go.mod"
+          cache: ${{ runner.arch != 'arm64' }}
```
Comment (poszu) [inline .github/workflows/ci.yml:66]: Why?

Comment (fasmat) [reply .github/workflows/ci.yml:66]: our selfhosted runners don't need caching - they keep the downloaded modules between builds anyway. But their cache is huge (several GiB) and this slows down the build -> the cache needs to be uploaded and downloaded during every build. Not caching speeds up builds significantly on our self hosted runners.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (fasmat) [reply .github/workflows/ci.yml:41]: For now I would leave it as it is. We can in the future define our own standardized jobs/steps in a dedicated repo and just import them into all others to get rid of this "code duplication" 🙂

**Review by poszu — APPROVED** [review]

**Pull request merged**